Add plugin: YouTube Video Summarizer

mbramani opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/mbramani/obsidian-yt-video-summarizer

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

mbramani commented:
Thank you for the feedback. I have addressed the styling issues by:

1. Moving all inline styles to a dedicated `styles.css` file
2. Implementing BEM methodology for CSS classes
3. Using Obsidian's CSS variables for theme compatibility
4. Removing all JavaScript-based styling

The changes can be seen in commit: `refactor(styles): migrate modals to use CSS classes`

Zachatoo commented:
[callback: () => this.handleSummarizeCommand()](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/main.ts#L71), [callback: () => this.handleSummarizeCommandWithPrompt()](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/main.ts#L78)
This should be a [editor command](https://docs.obsidian.md/Plugins/User+interface/Commands#Editor+commands)

[console.log('Loading plugin data...');](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/services/storage.ts#L30), [console.log('No saved data found, using defaults');](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/services/storage.ts#L41)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[.setName('Gemini API Key')](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/settings.ts#L36), [.setName('Gemini Model')](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/settings.ts#L47), [.setName('Summary Prompt')](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/settings.ts#L60), [.setName('Max Tokens')](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/settings.ts#L71), [.setName('Reset Settings')](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/a3c41492749eb218d7890cc58c03b182c5bfc075/src/settings.ts#L93)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

mbramani commented:
fixed
- Your manifest does not have the required version property.
- Your latest version number is not valid. Only numbers and dots are allowed.
- Unable to find a release with the tag undefined. Make sure that the version in your manifest.json file in your repo points to the correct Github Release.

joethei commented:
[name: 'Summarize YouTube Video',](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/main.ts#L69), [name: 'Summarize YouTube Video (With Prompt)',](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/main.ts#L94), [Watch Video](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/main.ts#L206), [Key Points](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/main.ts#L208), [Technical Terms](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/main.ts#L214)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[api](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/settings.ts#L36), [api](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/settings.ts#L37)
API is always capitalized.

[return url.includes('youtube.com/') || url.includes('youtu.be/');](https://github.com/mbramani/obsidian-yt-video-summarizer/blob/da29d6351494ad503700ae9a451454f7c3a75f6d/src/services/youtube.ts#L53)
You should add the `https://` at the start for both and use `startsWith` instead of `includes`, otherwise this check will also be true for `https://example.com/youtu.be`

mbramani commented:
Hi @joethei, I've addressed the feedback in this commit:

Changes made:
1. UI text consistency:
   - Fixed sentence casing in commands and headers
   - Changed "Watch Video" to "Watch video"
   - Updated "Key Points" to "Key points"
   - Changed "Technical Terms" to "Technical terms"

2. API capitalization:
   - Updated all instances to use "API" instead of "api"

3. URL validation:
   - Improved YouTube URL validation using `startsWith()`
   - Added `https://` prefix checks
   - Updated validation to: 
     ```typescript
     return (
         url.startsWith('https://www.youtube.com/') || 
         url.startsWith('https://youtu.be/')
     );
     ```

You can view the changes in [this commit](https://github.com/mbramani/obsidian-yt-video-summarizer/commit/91d1bc3716604803f6f65b07fda9c872d061813e).
Let me know if anything needs further adjustment!

joethei commented:
You missed two of the sentence case changes (number 2 and 3)

mbramani commented:
Hi @joethei,

Thank you for your patience with this.
I have addressed the requested changes. 
You can review them here: [commit link](https://github.com/mbramani/obsidian-yt-video-summarizer/commit/280d60396e64ab9a3c2f454a0a54fa85adf4dbef).
